controllers/install/armadaserver_controller_unit_test.go

```diff
@@ -0,0 +1,162 @@
+package install
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/armadaproject/armada-operator/apis/install/v1alpha1"
+	"github.com/armadaproject/armada-operator/test/k8sclient"
+
+	installv1alpha1 "github.com/armadaproject/armada-operator/apis/install/v1alpha1"
+)
+
+func TestArmadaServerReconciler_Reconcile(t *testing.T) {
```

Comment:
You can leave this as a followup but we added a few cases in executor and binoculars that may interest you.  

You have the happy path.  

There is one case where you try to reconcile a server object and it doesn't exist.  

The last one is when trying to delete the server.

Reply:
Good suggestion - I will add that in a subsequent PR shortly.